source/BadgeOverlaySupport.js

```diff
@@ -0,0 +1,220 @@
+	*
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', badgeSrc: 'assets/my-icon.png',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	* ```
+	*
+	* The item may define a
+	* [position]{@link moon.BadgeOverlaySupport#position} to
+	* specify position of badge. Possible values are 'top' or 'bottom'. Defaulted as 'bottom'.
+	* ```javascript
+	* {name: 'list', kind: 'moon.DataList', components: [
+	*	{mixins: ['moon.BadgeOverlaySupport'], badgeType: 'video', position: 'top',
+	*		kind: 'moon.ImageItem', bindings: [
+	*			{from: '.model.title', to: '.label'},
+	*			{from: '.model.description', to: '.text'},
+	*			{from: '.model.coverSource', to: '.source'}
+	*		]
+	*	}
+	* ]}
+	*
+	* @mixin moon.BadgeOverlaySupport
+	* @protected
+	*/
+	moon.BadgeOverlaySupport = {
+
+		/**
+		* @private
+		*/
+		name: 'moon.BadgeOverlaySupport',
+
+		/**
+		* @private
+		*/
+		classes: 'moon-badge-overlay-support',
+
+		/**
+		* Identifies the MIME type of badge.
+		* Possible values are "image", "audio" or "video".
+		* Defaulted as 'image'.
+		*
+		* @name moon.BadgeOverlaySupport#badgeType
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeType: null,
+
+		/**
+		* Font icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeIcon
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeIcon: null,
+
+		/**
+		* Url for icon to be used in place of default icon.
+		*
+		* @name moon.BadgeOverlaySupport#badgeSrc
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		badgeSrc: null,
+
+		/**
+		* Sepcifies the alignment position of badge.
+		* Possible values are 'top' or 'bottom'.
+		* Defaulted as 'bottom'.
+		*
+		* @name moon.BadgeOverlaySupport#position
+		* @type {String}
+		* @default undefined
+		* @public
+		*/
+		position: null,
+
+		/**
+		* @private
+		*/
+		events: {
+
+			/**
+			* {@link moon.BadgeOverlaySupport#onBadgeTap}
+			*/
+			onBadgeTap: '',
+
+			/**
+			* {@link moon.BadgeOverlaySupport#onBadgeIconTap}
+			*/
+			onBadgeIconTap: ''
+		},
+
+		/**
+		* @method
+		* @private
+		*/
+		create: enyo.inherit(function (sup) {
+			return function () {
+				this.kindComponents = enyo.Component.overrideComponents(this.kindComponents, this._badgeOverride);
+				sup.apply(this, arguments);
+				this.positionChanged();
+				this.badgeTypeChanged();
+				if (this.badgeSrc) {
+					this.$.badgeScrimIcon.set('icon','');
+				}
+			};
+		}),
+
+		/**
+		* @private
+		*/
+		bindings: [
+			{from: 'badgeSrc', to: '$.badgeScrimIcon.src'},
+			{from: 'badgeIcon', to: '$.badgeScrimIcon.icon', transform: function(val){ return val ? val : 'play';}}
+		],
+
+		/**
+		* @private
+		*/
+		_badgeOverride: {
+			image: { kind:"moon.Image", components:[{
```

Comment:
Please take a quick pass to convert double quotes to single and add spaces after `function`, `if`, and `switch` statements like so:

`method: function () {`, `if (something) {` and `switch (condition) {`